script.js

```diff
@@ -2,11 +2,104 @@
+    // Use innerHTML to add text to the anchor tag
+    h4TagElem.innerHTML =
+      `${episode.name}` +
+      `-S${episode.season.toString().padStart(2, "0")}
+        E${episode.number.toString().padStart(2, "0")}`;
+    pElem.innerHTML = episode.summary;
+    // Append the elements to the DOM
+    rootElem.appendChild(div);
+    div.append(h4TagElem, img, pElem);
+    // use setAttribute to add the card class to the div
+
+    div.setAttribute(
+      "class",
+      " main-wrapper card html body main-content card-block img"
+    );
+    //div.setAttribute("class", "cards")
+    img.setAttribute("src", episode.image.medium);
+  });
+}
+//filter episodes
+const searchBar = document.querySelector("input");
+searchBar.addEventListener("keyup", filterSearch);
+let main = document.querySelector("#root");
+function filterSearch(event) {
+  let firstValue = event.target.value.toLowerCase();
+  /* done with Orhan's help*/
+  let episodes = [...main.children]; // making episodes into an array
+  episodes.forEach((show) => {
+    const showText = show.innerText.toLowerCase();
+    if (showText.indexOf(firstValue) < 0) {
+      show.style.display = "none";
+    } else if (showText.indexOf(firstValue.length)) {
+      show.style.display = "inline-flex";
+    }
+  });
+  let filteredSearch = episodes.filter(
+    (ep) => getComputedStyle(ep).display == "inline-flex"
+  );
+  // Display search value
+  const displayCount = document.getElementById("search-result");
+  let results = (displayCount.innerText = `Displaying ${filteredSearch.length}/ ${episodes.length}`);
+}
+
+let episodes = [...main.children];
+//Show Episode Selections
+const selectElement = document.getElementById("select-episode");
+function selectEpisode(episodeList) {
+  episodeList.forEach((episode) => {
+    let option = document.createElement("option");
+    let episodeName = `S${episode.season
+      .toString()
+      .padStart(2, "0")}E${episode.number.toString().padStart(2, "0")}-${
+      episode.name
+    }`;
+    option.appendChild(document.createTextNode(episodeName));
+    option.appendChild(document.createTextNode("Please Select an Episode"));
+    option.value = "option value";
+    selectElement.appendChild(option);
+  });
+}
+selectElement.addEventListener("change", handleSelect);
+function handleSelect(event) {
+  let episodeNumber = event.target[0].textContent.split("-");
+  //console.log(e.target[0].textContent);
+  let episodes = [...main.children];
+  console.log(episodes[0].textContent);
+  episodes.forEach((episode) => {
+    let episodeText = episode.innerText.toLowerCase();
+    if (episodeText.indexOf(episodeNumber) < 0) {
+      episode.style.display = "none";
+    } else if (episodeText.indexOf(episodeNumber)) {
+      episode.style.display = "inline-flex";
+
+      let selectedEpisode = episode.filter(
+        (ep) => getComputedStyle(ep).display == "inline-flex"
+      );
+      console.log(selectedEpisode);
+    }
+  });
+
+  console.log(episodes[0]);
```

Comment:
You were close but there's a few issues:
- You should set the `option.value` on line 78 to the actual `episodeName`, not the hard-coded string 'option-value'. This will be the value that will be provided to you when the user selects something!
- In the `handleSelect`, you can then access that value, but also need to do some further manipulation to get it to work: `let episodeNumber = event.target.value.split('-')[0].toLowerCase();`
- the `else if` needs to be just `else` otherwise you're not catching the `0` case
- code on lines 95+ is not necessary

You should end up with something like this which works:

```
const selectElement = document.getElementById("select-episode");
function selectEpisode(episodeList) {
  episodeList.forEach((episode) => {
    let option = document.createElement("option");
    let episodeName = `S${episode.season
      .toString()
      .padStart(2, "0")} E${episode.number.toString().padStart(2, "0")}-${
      episode.name
      }`;
    option.appendChild(document.createTextNode(episodeName));
    option.appendChild(document.createTextNode("Please Select an Episode"));
    option.value = episodeName;
    selectElement.appendChild(option);
  });
}
selectElement.addEventListener("change", handleSelect);
function handleSelect(event) {
  let episodeNumber = event.target.value.split('-')[0].toLowerCase();
  //console.log(e.target[0].textContent);
  let episodes = [...main.children];

  episodes.forEach((episode) => {
    let episodeText = episode.innerText.toLowerCase();
    if (episodeText.indexOf(episodeNumber) < 0) {
      episode.style.display = "none";
    } else {
      episode.style.display = "inline-flex";
    }
  });
}
```

As I said above, using DOM as the source of truth is not ideal. Here you are comparing strings to decide which episode to show, which is prone to error (e.g. if you don't lower case it, it won't work, etc.) This is what led you to a tricky situation which was hard to debug. Instead if you have your 'source of truth' in a javascript object in your code, it will be much easier to do all the data manipulation (filtering, looping through, finding the correct one to display etc.)

Reply:
Further to that, it would be better if you use the show `id` value in the data to uniquely identify a show, not it's title or season / episode combination.